.github/workflows/main.yml

```
@@ -0,0 +1,103 @@
+name: Application Tests
+
+on:
+  pull_request:
+  push:
+
+jobs:
+  linting:
+    name: Run Linting
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Project
+        uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+      - name: Download Cached Gems
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: bundler-cache-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: |
+            bundler-cache-
+      - name: Install Dependencies
+        run: bundle install
```

```suggestion
        with:
          bundler-cache: true
```

The `ruby/setup-ruby` actions [recommends against caching manually](https://github.com/ruby/setup-ruby#caching-bundle-install-manually).

---

Thanks for the suggestion but I am not sold on using that over the documented Github caching mechanisms that we can configure/tune as we need.

---

Can you at least make sure whatever you setup works? Right now installation seems to be taking 5 minutes, which I don't think it's expected since `bundle install` has already run successfully several times on this branch.

My experience configuring catching of github actions is that `actions/cache` is quite hard to get right, which is exactly what the docs in `ruby/setup-ruby` I pointed to explain.

A one liner provided by ruby's official actions seems easier, but I'm happy with anything as long as it works.

---

@colby-swandale `ruby/setup-ruby` is defacto standard and there is no reason to not use built-in caching. @eregon can you confirm?

---

Yes.
I genuinely wonder why @colby-swandale (I've seen others as well) prefer to do a manual cache.
To be a clear a manual cache:
* is way more verbose
* is error prone, e.g. the key above does not include the operating system and version or the Ruby version
* is incorrect for -head versions
* using restore keys leads to extra gems in the cache

I'm thinking maybe I should enable bundler-cache by default in ruby/setup-ruby. Then maybe it would be clearer adding a manual cache is not the way?